app/assets/javascripts/Components/marks_spreadsheet.jsx

```diff
@@ -260,6 +260,7 @@ class RawMarksSpreadsheet extends React.Component {
           ref={(r) => this.actionBox = r}
           toggleRelease={this.toggleRelease}
           showHidden={this.state.show_hidden}
+          is_allowed_to_release_unrelease={this.props.is_allowed_to_release_unrelease}
```

Comment:
In general. prefer the prop name "can_*" rather than "is_allowed_to_*" since the former is shorter. So for example, here use `can_release_unrelease`. But also in this particular case, I think `can_release` is a better name (basically we're "merging" the permission of release/unrelease into a single name).

You should go through all of the new props you added and do this renaming.